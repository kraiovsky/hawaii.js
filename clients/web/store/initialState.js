export default {
  isFinished: {
    auth: false,
  },
  auth: {
    access_token: null,
    refresh_token: null,
    scope: null,
    uid: null,
    email: null,
  },
  pageTitle: '',
}
