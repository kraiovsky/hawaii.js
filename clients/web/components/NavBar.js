import React from 'react'
import Link from 'next/link'
import { useSelector } from 'react-redux'
import { getIsFinished, getIsAuthenticatedUser } from '../store/selectors'
import LoginSignup from './AuthNav'
import Logout from './Logout'

export default () => {
  const state = useSelector(state => state)

  return (
    <>
      <div>
        <Link href="/">
          <a>Home</a>
        </Link>
        {' | '}
        <Link href="/dashboard">
          <a>Dashboard</a>
        </Link>
        {getIsFinished(state, 'auth') && !getIsAuthenticatedUser(state) && (
          <>
            {' | '}
            <LoginSignup />
          </>
        )}
        {getIsFinished(state, 'auth') && getIsAuthenticatedUser(state) && (
          <>
            {' | '}
            <Logout />
          </>
        )}
      </div>
    </>
  )
}
