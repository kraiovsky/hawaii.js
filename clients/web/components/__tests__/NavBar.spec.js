import React from 'react'
import renderer from 'react-test-renderer'
import { useSelector } from 'react-redux'
import initialState from '../../store/initialState'
import NavBar from '../NavBar'

jest.mock('react-redux')

describe('Render NavBar', () => {
  test('Auth init in progress', () => {
    useSelector.mockImplementation(() => initialState)
    const component = renderer.create(<NavBar />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('👎 Auth init finished - unauthenticated user', () => {
    useSelector.mockImplementation(() => ({
      ...initialState,
      isFinished: {
        auth: true,
      },
    }))
    const component = renderer.create(<NavBar />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('👍 Auth init finished - authenticated user', () => {
    useSelector.mockImplementation(() => ({
      ...initialState,
      isFinished: {
        auth: true,
      },
      auth: {
        access_token: 'some-access-token',
        refresh_token: 'some-refresh-token',
        scope: 'super-user',
        uid: '123-user',
        email: 'user@email.com',
      },
    }))
    const component = renderer.create(<NavBar />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
