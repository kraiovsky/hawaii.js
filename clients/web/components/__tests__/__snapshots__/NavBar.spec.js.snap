// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NavBar Auth init in progress 1`] = `
<div>
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
   | 
  <a
    href="/dashboard"
    onClick={[Function]}
  >
    Dashboard
  </a>
</div>
`;

exports[`Render NavBar 👍 Auth init finished - authenticated user 1`] = `
<div>
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
   | 
  <a
    href="/dashboard"
    onClick={[Function]}
  >
    Dashboard
  </a>
   | 
  <a
    href="#"
    onClick={[Function]}
  >
    Log out
  </a>
</div>
`;

exports[`Render NavBar 👎 Auth init finished - unauthenticated user 1`] = `
<div>
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
   | 
  <a
    href="/dashboard"
    onClick={[Function]}
  >
    Dashboard
  </a>
   | 
  <a
    href="/auth"
    onClick={[Function]}
  >
    Login
  </a>
   | 
  <a
    href="/auth"
    onClick={[Function]}
  >
    Signup
  </a>
</div>
`;
